RecoTracker/TkSeedGenerator/plugins/DeepCoreSeedGenerator.cc

```diff
@@ -0,0 +1,583 @@
+    const PixelClusterParameterEstimator* pixelCPE,
+    const reco::Vertex& jetVertex,
+    int layer,
+    const edmNew::DetSetVector<SiPixelCluster>& clusters) {
+  std::vector<GlobalVector> clustDirs;
+  for (const auto& detset_int : clusters) {
+    const GeomDet* det_int = geometry_->idToDet(detset_int.id());
+    int lay = tTopo->layer(det_int->geographicalId());
+    if (lay != layer)
+      continue;  //NB: saved bigClusters on all the layers!!
+    for (const auto& aCluster : detset_int) {
+      GlobalPoint clustPos = det_int->surface().toGlobal(
+          pixelCPE->localParametersV(aCluster, (*geometry_->idToDetUnit(detset_int.id())))[0].first);
+      GlobalPoint vertexPos(jetVertex.position().x(), jetVertex.position().y(), jetVertex.position().z());
+      GlobalVector clusterDir = clustPos - vertexPos;
+      GlobalVector jetDir(jet.px(), jet.py(), jet.pz());
+      if (Geom::deltaR(jetDir, clusterDir) < deltaR_) {
+        clustDirs.emplace_back(clusterDir);
+      }
+    }
+  }
+  return clustDirs;
+}
+
+std::unique_ptr<DeepCoreCache> DeepCoreSeedGenerator::initializeGlobalCache(const edm::ParameterSet& iConfig) {
+  // this method is supposed to create, initialize and return a DeepCoreCache instance
+  std::unique_ptr<DeepCoreCache> cache = std::make_unique<DeepCoreCache>();
+  std::string graphPath = iConfig.getParameter<edm::FileInPath>("weightFile").fullPath();
+  cache->graph_def = tensorflow::loadGraphDef(graphPath);
+  return cache;
+}
+
+void DeepCoreSeedGenerator::globalEndJob(DeepCoreCache* cache) { delete cache->graph_def; }
+
+// ------------ method called once each job just before starting event loop  ------------
+void DeepCoreSeedGenerator::beginJob() {}
+
+// ------------ method called once each job just after ending the event loop  ------------
+void DeepCoreSeedGenerator::endJob() {}
+
+// ------------ method fills 'descriptions' with the allowed parameters for the module  ------------
+void DeepCoreSeedGenerator::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  edm::ParameterSetDescription desc;
+  desc.add<edm::InputTag>("vertices", edm::InputTag("offlinePrimaryVertices"));
+  desc.add<edm::InputTag>("pixelClusters", edm::InputTag("siPixelClustersPreSplitting"));
+  desc.add<edm::InputTag>("cores", edm::InputTag("jetsForCoreTracking"));
+  desc.add<double>("ptMin", 300);
+  desc.add<double>("deltaR", 0.1);
+  desc.add<double>("chargeFractionMin", 18000.0);
+  desc.add<double>("centralMIPCharge", 2);
+  desc.add<std::string>("pixelCPE", "PixelCPEGeneric");
+  desc.add<edm::FileInPath>("weightFile",
+                            edm::FileInPath("RecoTracker/TkSeedGenerator/data/DeepCoreSeedGenerator_TrainedModel.pb"));
+  desc.add<std::vector<std::string>>("inputTensorName", {"input_1", "input_2", "input_3"});
+  desc.add<std::vector<std::string>>("outputTensorName", {"output_node0", "output_node1"});
+  desc.add<double>("probThr", 0.85);
+  descriptions.addDefault(desc);
```

Comment:
this should not be there.
I think you need something like:
```
descriptions.add("DeepCoreSeedGenerator", desc);
``` 

or the like (depending on how you use this elsewhere)

Reply:
I would propose:
```diff
-  descriptions.addDefault(desc);
+  descriptions.addWithDefaultLabel(desc);
 }
```

I don't know what's the difference between `add` and `addDefault`, but the latter does not seem correct.

Reply:
Thank you for your comment Marco. I followed the default template. Actually in the rest of the code is imported called DeepCoreSeedGenerator_cfi, so I would like to have a description which allows continuing to use DeepCoreSeedGenerator_cfi import.  
like here: https://github.com/cms-sw/cmssw/pull/32222/files#diff-f3743f96f0615690ff1592df517f73afa90d9e5b0f252151df6beafe8efbde09R145

Reply:
`descriptions.add("DeepCoreSeedGenerator", desc);` will create `DeepCoreSeedGenerator_cfi.py`

Reply:
ok I'll fix in this way

Reply:
FYI, I think the template by `mkedprod` is outdated and needs to be updated, I opened an issue: https://github.com/cms-sw/cmssw/issues/32282

Reply:
lower case style for instances/objects still applies here, together with a style to match the module instance with the file name; so, it's `deepCoreSeedGenerator` and addWithDefaultLabel is preferred

Reply:
@slava77 apart from the _cfi/description are you suggesting to change the name from DeepCoreSeedGenerator to deepCoreSeedGenerator everywhere? (I have to say that I always used, in all the circumstances, DeepCore with the capital letter, inspired by other "DeepStuff" of CMS, like DeepMET)
https://github.com/cms-sw/cmssw/blob/master/RecoMET/METPUSubtraction/plugins/DeepMETProducer.cc

Reply:
the class name of the producer is DeepCore..., the module running in the jobs (an instance/object of the module class) is deepCore...

Reply:
ok understood and fixed with deepCore for the module and DeepCore for the class, like DeepMET